runners/spark/src/main/java/org/apache/beam/runners/spark/translation/MultiDoFnFunction.java

```diff
@@ -151,7 +151,8 @@ public TimerInternals timerInternals() {
     return new SparkProcessContext<>(
         doFn, doFnRunnerWithMetrics, outputManager,
         stateful ? new TimerDataIterator(timerInternals) :
-            Collections.<TimerInternals.TimerData>emptyIterator()).processPartition(iter);
+            Collections.<TimerInternals.TimerData>emptyIterator()).processPartition(iter)
```

Comment:
So doing it this way will prevent Spark from being able to spill to disk, which is not great. It doesn't have the be fixed in this, but we should make a JIRA to follow up and audit all of the places we switched from Iterable to Iterator return and make sure we clean up the ones that matter later on.

Reply:
Holden would you mind to create the JIRA and explain what is the correct expected way this should be done if possible. Thanks.

Reply:
https://issues.apache.org/jira/browse/BEAM-3290

Reply:
Agree. The purpose for this PR is the update first. For sure,  we saw potential improvements.  However,  I think it makes sense to do these improvements after Spark 2 upgrade.

Reply:
Thanks @holdenk !